Nextcloud support

tostc opened the pull request:
I've implemented the feature request of Issue #20. You can now backup your world to nextcloud. I've generalize some texts, for example

> "An error occurred while uploading to Dropbox"

is now

> "An error occurred while uploading to {Placeholder}"

- There is now an enum which contains the cloud type(Dropbox, Nextcloud). This is used for the placeholder
- I've move the Dropbox package to a sub package clouds which contains also a Nextcloud package. - The callback class for Dropbox `DropboxActivity` is now named `CloudActivity` and is also used by Nextcloud. It's also in the cloud package.
- Because the Nextcloud library was added I needed to enable multidex support for the app, I hope there are no problems with that.
- Added a login Activity for Nextcloud, there you can set the server address your username and password.
- Because the Nextcloud library doesn't support OAuth2, I never save the password on flash.

Hope I haven't forget anything. The new texts for Nextcloud are translated to german and english.

P.S.: Maybe Owncloud will also work, but this is not tested.

tostc committed:
Support for Nextcloud

- Added support for nextcloud.
- Added login panel for nextcloud
- Added translations for nextcloud
- Added new menu entry for nextcloud
- Makes the dropbox types more generic for another clouds supports

tostc committed:
Refactoring

- Moved the nexcloud and dropbox package to clouds.
- Generalized DropboxActivity to CloudActivity.
- Refactor CloudActivity callbacks to general names

MarquisLP committed:
Add missing Manifest property for Multidex support

MarquisLP committed:
Show example Nextcloud server URL to prevent confusion

MarquisLP commented:
Thanks again for the help! This issue has been sitting in the backlog for quite some time, so I really appreciate you having taken it on.

Just now, I've pushed two changes to your branch:
- I've added `android:name="androidx.multidex.MultiDexApplication"` to the AppManifest, as stated in the [docs](https://developer.android.com/studio/build/multidex#mdex-gradle). Without this, the app would not build.
- I've edited the "Nextcloud Server Address" label to include an example URL, so that users don't have to guess at the exact format.

Other than that, everything works well; I'm able to get the full upload process to work on my end.

One thing I have noticed, though, is that World folders are uploaded to the user's root folder on Nextcloud. Do you know if it would be possible to group all World folders under a parent "WorldScribe" folder, like what we do for Dropbox?

(This isn't a critical issue, but more of a nice-to-have. I'll still merge the branch even if it isn't possible.)

tostc committed:
Added grouped root folder

- All worlds are now in a folder named World Scribe in nextcloud.
- Translated "e.g" to german.

tostc commented:
Yes it's possible to group all worlds to a folder like Dropbox. I've added the group function.

MarquisLP commented:
Thanks, works like a charm!
